after upgrade router-plugin  to v1.121.22, cannot compile successfully project based on the file-based routing with Rspack or Rsbuild

### Which project does this relate to?

Router

### Describe the bug

before (compile successfully):
@tanstack/react-router: 1.116.0
@tanstack/router-plugin: 1.116.1
@rsbuild/core: 1.1.8


after: 
@tanstack/react-router: 1.121.21
@tanstack/router-plugin: 1.121.22
@rsbuild/core: 1.1.8

error:
```
error   Compile error: 
Failed to compile, check the errors for troubleshooting.
× Error: [html-rspack-plugin]: Child compilation failed:
  │   × Module parse failed:
  │   ╰─▶   × JavaScript parsing error: Expression expected
  │          ╭─[1:0]
  │        1 │ <!DOCTYPE html>
  │          · ─
  │        2 │ <!--
  │        3 │     NOTE: This HTML file is a template. If you open it directly in the browser,
  │          ╰────
  │ 
  │   help:
  │         File was processed with these loaders:
  │          * checkoutroot/node_modules/@rsbuild/core/compiled/html-rspack-plugin/loader.js
  │          * checkoutroot/node_modules/unplugin/dist/rspack/loaders/transform.js??tanstack-router:hmr
  │         You may need an additional loader to handle the result of these loaders.
```

### Your Example Website or App

reproduce [demo](https://github.com/qiuzhangxi77/quickstart-rspack-file-based-complie-failed-demo)

### Steps to Reproduce the Bug or Issue

see description

### Expected behavior

please see description

### Screenshots or Videos

_No response_

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]


### Additional context

_No response_

My Cloudflare deploy is actually failing, app builds but then dies after with:

```
17:03:08.197 | ✘ [ERROR] Error: Pages only supports files up to 25 MiB in size
-- | --
17:03:08.197 |  
17:03:08.197 | @rspack/binding-linux-x64-gnu/rspack.linux-x64-gnu.node is 59.7 MiB in size
```

--- a/packages/router-plugin/src/core/route-autoimport-plugin.ts
+++ b/packages/router-plugin/src/core/route-autoimport-plugin.ts
@@ -21,6 +21,8 @@ export const unpluginRouteAutoImportFactory: UnpluginFactory<
 
     transform: {
       filter: {
+        // this is necessary for webpack / rspack to avoid matching .html files
+        id: /\.(m|c)?(j|t)sx?$/,
         code: /createFileRoute\(|createLazyFileRoute\(/,
       },
       handler(code, id) {

--- a/packages/router-plugin/src/core/router-code-splitter-plugin.ts
+++ b/packages/router-plugin/src/core/router-code-splitter-plugin.ts
@@ -180,6 +180,8 @@ export const unpluginRouterCodeSplitterFactory: UnpluginFactory<
         filter: {
           id: {
             exclude: tsrSplit,
+            // this is necessary for webpack / rspack to avoid matching .html files
+            include: /\.(m|c)?(j|t)sx?$/,
           },
           code: 'createFileRoute(',
         },

--- a/packages/router-plugin/src/core/router-hmr-plugin.ts
+++ b/packages/router-plugin/src/core/router-hmr-plugin.ts
@@ -1,5 +1,4 @@
 import { generateFromAst, logDiff, parseAst } from '@tanstack/router-utils'
-import { getConfig } from './config'
 import { routeHmrStatement } from './route-hmr-statement'
 import { debug } from './utils'
 import type { Config } from './config'
@@ -12,16 +11,14 @@ import type { UnpluginFactory } from 'unplugin'
  */
 export const unpluginRouterHmrFactory: UnpluginFactory<
   Partial<Config> | undefined
-> = (options = {}) => {
-  let ROOT: string = process.cwd()
-  let userConfig = options as Config
-
+> = () => {
   return {
     name: 'tanstack-router:hmr',
     enforce: 'pre',
-
     transform: {
       filter: {
+        // this is necessary for webpack / rspack to avoid matching .html files
+        id: /\.(m|c)?(j|t)sx?$/,
         code: 'createFileRoute(',
       },
       handler(code, id) {
@@ -44,23 +41,6 @@ export const unpluginRouterHmrFactory: UnpluginFactory<
         }
         return result
       },
-    },
-
-    vite: {
-      configResolved(config) {
-        ROOT = config.root
-        userConfig = getConfig(options, ROOT)
-      },
-    },
-
-    rspack() {
-      ROOT = process.cwd()
-      userConfig = getConfig(options, ROOT)
-    },
-
-    webpack() {
-      ROOT = process.cwd()
-      userConfig = getConfig(options, ROOT)
     },
   }
 }

